frontend/templates/product/sections/ProductSection/ProductOptions.tsx

```diff
@@ -113,19 +111,60 @@ enum SelectorType {
    IMAGE_RADIO,
 }
 
-type Variant = Product['variants'][0];
-
 type Option = Product['options'][0];
 
 function findVariant(
-   variants: Variant[],
-   selectedOptions: Record<string, string>
+   product: Product,
+   previousOptions: Record<string, string>,
+   optionsUpdate: Record<string, string>
```

Comment:
Is it possible to ever update more than one option at a time? If not, I might prefer an `Option` to make the logic easier to follow.

```suggestion
   optionUpdate: Option
```

Reply:
At the moment no, it is not. 
No problem updating it

Reply:
Ah, I didn't realize `Option` was a different type than `{name: string, value: string}`. I like your idea to use `optionUpdate: {name: string, value: string}`.